lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -379,6 +379,36 @@ class DecomposeAtenMatmulOp : public OpRewritePattern<AtenMatmulOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenTOp : public OpRewritePattern<AtenTOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenTOp op,
+                                PatternRewriter &rewriter) const override {
+    Value lhs = op.self();
+    int lhsRank = getTensorRank(lhs);
+    auto loc = op.getLoc();
+
+    if (lhsRank < 2)
+      rewriter.replaceOp(op, lhs);
+    else if (lhsRank > 2 || lhsRank < 0) {
```

Comment:
the `lhsRank < 0` case will never reach here because it will go into the first `if` branch above. You should check that the rank is valid right after `getTensorRank` before doing anything else